### PR TITLE
clouds: Fix the clouds.yaml locations

### LINKED ISSUE #3038: clouds.Parse() ignores the local clouds.yaml
The new function `clouds.Parse()` picks the clouds.yaml locations in a reverse order, that is:
1. `/etc/openstack/clouds.yaml`
1. `~/.config/openstack/clouds.yaml`
1. `./clouds.yaml`

CC @itzikb-redhat

### PR DESCRIPTION
**Pull request by pierreprinetti**
Before this patch, all default locations were scanned for a `clouds.yaml` even when a match was found. The last found `clouds.yaml` replaced any previous finding, effectively reversing the order of priority for the locations.

Fixes #3038

### CODE REVIEW

**Comment by pierreprinetti** [comment]
I am also removing the check on `*os.PathError` because according to the docs, [that's the only error that can ever come from `os.Open()`](https://pkg.go.dev/os#Open).

File: openstack/config/clouds/clouds.go [hunk 1]
```diff
@@ -90,27 +89,22 @@ func Parse(opts ...ParseOption) (gophercloud.AuthOptions, gophercloud.EndpointOp
 		}
 
 		for _, cloudsPath := range options.locations {
-			var errNotFound *os.PathError
 			f, err := os.Open(cloudsPath)
-			if err != nil && !errors.As(err, &errNotFound) {
-				return gophercloud.AuthOptions{}, gophercloud.EndpointOpts{}, nil, fmt.Errorf("failed to open %q: %w", cloudsPath, err)
+			if err != nil {
```
Comment (EmilienM) [inline openstack/config/clouds/clouds.go:93]: random question (mostly to fulfill my curiosity), why didn't you do:

```golang
f, _ := os.Open(cloudsPath)
defer f.Close()
options.cloudsyamlReader = f
```

Comment (EmilienM) [reply openstack/config/clouds/clouds.go:93]: Nevermind my comment, I misread the next lines.

**Review by EmilienM — APPROVED** [review]
LGTM

**Pull request merged**